pkg/virt-controller/network/ipamclaims.go

```diff
@@ -0,0 +1,206 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	virtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/namescheme"
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+
+	"kubevirt.io/client-go/kubecli"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+
+	ipamclaims "github.com/k8snetworkplumbingwg/ipamclaims/pkg/crd/ipamclaims/v1alpha1"
+)
+
+type NetConf struct {
+	AllowPersistentIPs bool `json:"allowPersistentIPs,omitempty"`
+}
+
+func GetNetworkToIPAMClaimName(virtClient kubecli.KubevirtClient, namespace string, name string, networks []virtv1.Network, persistentIPsEnabled bool) (map[string]string, error) {
```

Comment:
if persistentIPsEnabled is false we should not even call this functions so no need to pass it as a flag.

Reply:
We want to error but in case there are NAD that requested persistentIPs right ?
if so we need to check something like
```
	if !persistentIPsEnabled && len(networkToIPAMClaimName) != 0 {
		return nil, fmt.Errorf("persistenIPs is requested but PersistentIPsGate is disabled")
	}
```
so instead doing it on every caller, i did it once inside, wdyt?